server/index.js

```diff
@@ -47,6 +47,86 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) return [];
+  const sql = `
+    select "c"."cartItemId",
+       "c"."price",
+       "p"."productId",
+       "p"."image",
+       "p"."name",
+       "p"."shortDescription"
+  from "cartItems" as "c"
+  join "products" as "p" using ("productId")
+ where "c"."cartId" = $1
+  `;
+  const value = [req.session.cartId];
+  db.query(sql, value)
+    .then(result => res.json(result.rows))
+    .catch(err => next(err));
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const pattern = /^[1-9]\d*$/;
+  if (pattern.exec(req.body.productId) === null) {
+    next(new ClientError(`Requested product (id: ${req.body.productId}) is not a positive integer.`, 400));
+  }
+  const getPriceSql = `
+    select "price"
+      from "products"
+     where "productId" = $1;
+  `;
+  const getPriceValue = [req.body.productId];
+  db.query(getPriceSql, getPriceValue)
+    .then(result => {
+      if (!result.rows[0]) {
+        next(new ClientError(`Requested product (id: ${req.body.productId}) does not exist in the database.`, 404));
```

Comment:
You should `throw` this error or `return Promise.reject(new ClientError(/* etc */))` to shortcut to your `catch()` callback.